l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py

```diff
@@ -1888,8 +1881,52 @@ def importFatturaPA(self):
                     )
                     invoice.write({"tax_representative_id": tax_partner_id})
                 if Intermediary:
-                    Intermediary_id = self.getPartnerBase(Intermediary.DatiAnagrafici)
-                    invoice.write({"intermediary": Intermediary_id})
+                    vat = False
+                    nome = Intermediary.DatiAnagrafici.Anagrafica.Nome or ""
+                    cognome = Intermediary.DatiAnagrafici.Anagrafica.Cognome or ""
+                    denominazione = (
+                        Intermediary.DatiAnagrafici.Anagrafica.Denominazione or ""
+                    )
+
+                    if Intermediary.DatiAnagrafici.IdFiscaleIVA:
+                        partner = self.getPartnerBase(
+                            Intermediary.DatiAnagrafici, partnerCreate=False
+                        )
+
+                        if isinstance(partner, int):
+                            partner_id = self.env["res.partner"].browse(partner)
+                            invoice.write({"intermediary": partner_id})
```

Comment:
Qui mi pare ci sia un po' di confusione tra l'ID del partner (un intero) e un record del partner perché non si sa bene cosa potrà uscire dalla nuova implementazione di `getPartnerBase`.

In generale penso sia più chiaro usare il suffisso `_id` quando si parla di un ID (es `partner_id`), e nessun suffisso quando si parla di un record (es `partner`).

Quindi suggerirei
```suggestion
                        partner_id = self.getPartnerBase(
                            Intermediary.DatiAnagrafici, partnerCreate=False
                        )

                        if isinstance(partner_id, int):
                            invoice.write({"intermediary": partner_id})
```
che è anche più simile al codice prima di questa PR (era `intermediary_id`).

Reply:
Si hai ragione in questo caso non ci avevo fatto caso

Reply:
> Però in questo caso `getPartnerBase` può restituire o in intero (se trova il partner) oppure un dizionario che verra sfruttato nella funzione di import. Per questo avevo pensato di generalizzare e scrivere partner. Infatti c'è anche la seconda condizione sotto che valuta se viene restituito un dizionario

Ok, però dentro `partner` di solito c'è un record; se non è così si crea confusione nel resto del codice.
Ad esempio
```py
partner_id = self.env["res.partner"].browse(partner)
```
è il contrario di quello che si farebbe normalmente.
Se non si sa di che tipo sia la variabile puoi chiamarla tipo `partner_id_or_vals`.

Ho come la sensazione che questo metodo che può restituire un dizionario o un intero ci stia complicando la vita, forse sarebbe più semplice qualcosa del tipo:
```py
partner_id = self._find_partner(...)
if partner_id:
    invoice.write(...)
else:
    partner_values = self._prepare_partner_values(...)
```
Cosa ne pensi?